### ISSUE TITLE
create_mscoco_tf_record.py for Object Detection API

### ISSUE THREAD

**Issue by BoltzmannBrain** [tensorflow/models#1710]
The new [Object Detection API](https://github.com/tensorflow/models/tree/master/object_detection) is great, but how come there are no details for evaluating the models on MS COCO? The models were trained on this data, so it would be great to include a create_mscoco_tf_record.py (like you have for the PASCAL and Oxford pet datasets) to test on the MS COCO data. Not to mention then people would be able to reproduce the paper results.

**Comment by BoltzmannBrain** [comment]
There's a version for the [im2txt model](https://github.com/tensorflow/models/blob/master/im2txt/im2txt/data/build_mscoco_data.py), but it's specific to the MS COCO captions dataset.

**Comment by jch1** [comment]
Hi @BoltzmannBrain - yes, we'd like to have scripts for COCO.  For the first release, we decided to leave it out for a number of reasons, but if you don't want to wait, it should be easy enough to modify the code for the im2txt model to output tfrecord files for bounding box detection.  

The config files that we have provided for the Pets dataset are exactly the same ones (except for the paths to labelmap and dataset) that we use internally for training on COCO, so really the main thing that's missing is a dataset.

**Comment by BoltzmannBrain** [comment]
Sounds good, I'll work off the im2txt version for now. Thank you for the quick response :)

**Comment by jaydee713** [comment]
@BoltzmannBrain Did you end up creating a script? trying to evaluate the model on MSCOCO as well

**Comment by BoltzmannBrain** [comment]
@jaydee713 not yet, hopefully I'll have time to get to it this weekend. If I do I'll be sure to push it and let you know here.

**Comment by jaydee713** [comment]
@BoltzmannBrain  any luck? I've tried fidgeting with a script myself but keep running into miscellaneous errors... I'm still relatively new to TF so it's giving me quite the headache haha. Hopefully you've had better luck?

**Comment by BoltzmannBrain** [comment]
@jaydee713 sorry this has fallen to my back-burner; it's one of a few experiments for a side project. Not to mention that it looks like I'll need to train my own models and can't utilize the pre-trained weights here 🙁 .

**Comment by ali01** [comment]
Closing this for now.

**Comment by MetaPeak** [comment]
@jaydee713  I wrote a script to convert coco dataset to tfrecords  according the "create_pascal_tf_record.py", actually, it is very easy. You only need to supply the labels of bbox, coordinates of the bboxs. If you need it, you can contact me.

**Comment by MetaPeak** [comment]
Hi, everyone, I wrote the code to convert coco dataset to tf record for Tensorflow object detection API . Here is the code.
https://github.com/MetaPeak/tensorflow_object_detection_create_coco_tfrecord

**Comment by offbye** [comment]
Great job!   I will try it immediately

**Comment by offbye** [comment]
I just have try it and fix some problems! Now  I can train model on  mscoco dataset !  
https://github.com/offbye/tensorflow_object_detection_create_coco_tfrecord/

**Comment by janeliulwq** [comment]
FYI, https://github.com/tensorflow/models/blob/master/research/object_detection/dataset_tools/create_coco_tf_record.py was added in a recent update :), and coco_detection_metrics was added too!

**Comment by GustavZ** [comment]
Doe someone of you know how to create the tf record file for only some selected, but not all, coco classes?

**Comment by MyyyPanda** [comment]
> Doe someone of you know how to create the tf record file for only some selected, but not all, coco classes?

hello GustavZ, have you found any way to create part of coco dataset into TFrecord file?

**Comment by NoraKrbr** [comment]
> > Doe someone of you know how to create the tf record file for only some selected, but not all, coco classes?
> 
> hello GustavZ, have you found any way to create part of coco dataset into TFrecord file?

I would also be interested if someone managed to create part of the coco dataset.
